### ISSUE TITLE
Monitor iptables rule

### ISSUE THREAD

**Issue by tazjin** [jtblin/kube2iam#9]
When kube2iam is configured to set up the iptables rule I think it should also monitor the rule's presence somehow. I for one am afraid of some process resetting the rule and thus allowing pods some dangerous privilege escalations.

Thoughts on this? If people think it's a good idea I may be able to spend some time on implementing it.

**Comment by mgoodness** [comment]
I can see this being valuable. Should kube2iam also remove the rule when it's de-scheduled from a node? Might be important if/when #8 is implemented.

**Comment by tazjin** [comment]
I'm not sure if it's possible to remove it reliably (what if kube2iam crashes / gets killed for some reason?). I think in terms of worst-case scenarios failing credential lookups may be better than privilege escalations.

**Comment by rickypai** [comment]
related commit of how Amazon ECS handles it: https://github.com/aws/amazon-ecs-init/commit/04d5735d8b9b5a81e55ae87971b196bbc5e2944c

although that might be only done during instance initialization, and would not solve the case corrupted routing tables.

**Comment by mgoodness** [comment]
@tazjin have you put any work into this? If not, I may tackle it over the next few days. As far as removing the rule, I'm thinking we could implement it as a [PreStop hook](http://kubernetes.io/docs/user-guide/container-environment/).

**Comment by tazjin** [comment]
@mgoodness Nope, not yet. I'm currently tracking issue #8 as well because it may be very relevant to let the masters deal with running `kube2iam` such that the nodes don't need to have the `sts:AssumeRole` capability.

**Comment by jtblin** [comment]
**Proposed solution for monitoring iptables rules:**

To ensure the persistence of the required NAT rules, we propose introducing a background monitoring routine. This would prevent cases where other processes or manual resets inadvertently remove the `kube2iam` rules.

**Technical Details of the Proposal:**
- **Background Routine**: A lightweight goroutine would be added to the `Server` that periodically verifies the existence of the NAT rule (e.g., using `iptables -C`).
- **Automatic Recovery**: If the rule is found to be missing, the routine would automatically re-apply it using the existing `iptables.AddRule` logic.
- **Configurable Frequency**: A new flag, `--iptables-sync-period`, would allow users to control how often this check runs (defaulting to **60 seconds**).
- **Performance Impact**: The impact is negligible, as it involves a single, very fast command execution per period.

This monitoring would be optional and could be enabled via a flag. We welcome feedback on this approach and invite anyone interested in implementing this to contribute a PR.
